src/ts/components/extensions/richtexteditor/fragments/RichTextEditor.tsx

```diff
@@ -203,14 +220,39 @@ const RichTextEditor = ({
     };
     // Construct the toolbar. NB: Can't be updated after the editor is created.
     let mantineToolbar = undefined;
+
     // If any extensions are specified, load them. NB: Can't be changed after the editor is created.
-    const mantineExtensions = extensions.map((ext) => {
+
+    // Ensure CodeBlockLowlight always receives the JS `lowlight` instance,
+    // because Python users can't pass it through the Dash props.
+    const normalizedExtensions = extensions.map((ext) => {
+        // Case 1: String extension (e.g. "CodeBlockLowlight")
+        if (typeof ext === 'string') {
+            if (ext === 'CodeBlockLowlight') {
+                return { CodeBlockLowlight: { lowlight } };
+            }
+            return ext;
+        }
+
+        // Case 2: Object extension (e.g. { CodeBlockLowlight: {} })
+        const name = Object.keys(ext)[0];
+        const opts = ext[name];
+
+        if (name === 'CodeBlockLowlight') {
+            // Merge user options with lowlight
+            return { CodeBlockLowlight: { lowlight, ...opts } };
+        }
```

Comment:
I see this `Object.keys(ext)[0]` pattern below as well, but it seems weird - what if `{}` was passed in? Or what if multiple extensions were passed in as a single object?

Cleaner might be something like:
```js
if ('CodeBlockLowlight' in ext) {
    return {...ext, CodeBlockLowlight: { lowlight, ...ext.CodeBlockLowlight } };
}
```
(or omit the `...ext,` at the beginning if you're confident there will only be one key)

Also below is it acceptable that using an unrecognized name will result in a JS error, from the line `extensionMap[name].configure(ext[name])`? It'll likely be something cryptic like "TypeError: undefined is not an object".